packages/venia-ui/lib/components/RichContent/PageBuilder/ContentTypes/Tabs/tabs.js

```diff
@@ -0,0 +1,308 @@
+    const [gradient, setGradient] = useState(null);
+    const isScrolling = useRef(false);
+    const clientX = useRef(0);
+    const scrollX = useRef(0);
+
+    const {
+        tabNavigationAlignment = 'left',
+        minHeight,
+        defaultIndex = 0,
+        headers = [],
+        textAlign,
+        border,
+        borderColor,
+        borderWidth,
+        borderRadius,
+        marginTop,
+        marginRight,
+        marginBottom,
+        marginLeft,
+        paddingTop,
+        paddingRight,
+        paddingBottom,
+        paddingLeft,
+        cssClasses = [],
+        children
+    } = props;
+
+    const onMouseDown = useCallback(event => {
+        isScrolling.current = true;
+        clientX.current = event.clientX;
+    }, []);
+
+    const onMouseUp = useCallback(() => {
+        isScrolling.current = false;
+    }, []);
+
+    const onMouseMove = useCallback(
+        event => {
+            if (isScrolling.current && scrollElement) {
+                scrollElement.scrollLeft =
+                    scrollX.current + (clientX.current - event.clientX);
+                scrollX.current = scrollElement.scrollLeft;
+                clientX.current = event.clientX;
+            }
+        },
+        [scrollElement]
+    );
+
+    const handleScroll = useCallback(
+        event => {
+            const navScrollElement = event.target;
+            if (navScrollElement.scrollLeft > 0) {
+                // If we've scrolled to the end of the scrollable element we only display a left gradient
+                if (
+                    navScrollElement.scrollLeft +
+                        navScrollElement.offsetWidth +
+                        1 >=
+                    navScrollElement.scrollWidth
+                ) {
+                    setGradient('left');
+                } else {
+                    // While scrolling we show gradients on both sides
+                    setGradient('both');
+                }
+            } else {
+                setGradient('right');
+            }
+        },
+        [setGradient]
+    );
+
+    useEffect(() => {
+        let navScrollElement;
+        const navigationWrapper = navigationRef.current;
+
+        if (
+            navigationWrapper &&
+            navigationWrapper.childNodes[0].nodeName === 'UL'
+        ) {
+            navScrollElement = navigationWrapper.childNodes[0];
+            setScrollElement(navScrollElement);
+            // If there are additional tabs hidden by scroll we display a gradient on the right
+            if (navScrollElement.scrollWidth > navScrollElement.offsetWidth) {
+                setGradient('right');
+            }
+            navScrollElement.addEventListener('scroll', handleScroll);
+        }
+
+        return () => {
+            if (navScrollElement) {
+                navScrollElement.removeEventListener('scroll', handleScroll);
+            }
+        };
+    }, [navigationRef, handleScroll]);
```

Comment:
The linter should have warned you about this, but refs don't belong in `dependencies`.

- Refs are objects that never change (instead, their private value property changes), so they would never cause a re-render.
- Even `ref.current` doesn't belong as a dependency, since it's a getter.
- React recognizes refs, so it knows not to flag a ref as a missing dependency.

```suggestion
    }, [handleScroll]);
```

Reply:
I got a little confused by `useRef` so thanks for the detailed explanation. Resolved: https://github.com/magento/pwa-studio/pull/1872/commits/50cbc1f9d2749c2067d71b61e71393336208f5b1